pkg/tsdb/cloudwatch/log_actions.go

```diff
@@ -0,0 +1,318 @@
+	if err != nil {
+		return nil, fmt.Errorf("Error: Parameter 'logGroupName' is required")
+	}
+	queryRequest.SetLogGroupName(logGroupName)
+
+	logStreamName, err := parameters.Get("logStreamName").String()
+	if err != nil {
+		return nil, fmt.Errorf("Error: Parameter 'logStream' is required")
+	}
+	queryRequest.SetLogStreamName(logStreamName)
+
+	if startTime, err := parameters.Get("startTime").Int64(); err == nil {
+		queryRequest.SetStartTime(startTime)
+	}
+
+	if endTime, err := parameters.Get("endTime").Int64(); err == nil {
+		queryRequest.SetEndTime(endTime)
+	}
+
+	logEvents, err := logsClient.GetLogEventsWithContext(ctx, queryRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	messages := make([]*string, 0)
+	timestamps := make([]*int64, 0)
+
+	sort.Slice(logEvents.Events, func(i, j int) bool {
+		return *(logEvents.Events[i].Timestamp) > *(logEvents.Events[j].Timestamp)
+	})
+
+	for _, event := range logEvents.Events {
+		messages = append(messages, event.Message)
+		timestamps = append(timestamps, event.Timestamp)
+	}
+
+	timestampField := data.NewField("ts", nil, timestamps)
+	timestampField.SetConfig(&data.FieldConfig{Title: "Time"})
+
+	messageField := data.NewField("line", nil, messages)
+
+	return data.NewFrame("logEvents", timestampField, messageField), nil
+}
+
+func (e *CloudWatchExecutor) handleDescribeLogGroups(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json) (*data.Frame, error) {
+	logGroupNamePrefix := parameters.Get("logGroupNamePrefix").MustString("")
+	var response *cloudwatchlogs.DescribeLogGroupsOutput = nil
+	var err error
+
+	if len(logGroupNamePrefix) < 1 {
+		response, err = logsClient.DescribeLogGroupsWithContext(ctx, &cloudwatchlogs.DescribeLogGroupsInput{
+			Limit: aws.Int64(parameters.Get("limit").MustInt64(50)),
+		})
+	} else {
+		response, err = logsClient.DescribeLogGroupsWithContext(ctx, &cloudwatchlogs.DescribeLogGroupsInput{
+			Limit:              aws.Int64(parameters.Get("limit").MustInt64(50)),
+			LogGroupNamePrefix: aws.String(logGroupNamePrefix),
+		})
+	}
+
+	if err != nil || response == nil {
+		return nil, err.(awserr.Error)
+	}
+
+	logGroupNames := make([]*string, 0)
+	for _, logGroup := range response.LogGroups {
+		logGroupNames = append(logGroupNames, logGroup.LogGroupName)
+	}
+
+	groupNamesField := data.NewField("logGroupName", nil, logGroupNames)
+	frame := data.NewFrame("logGroups", groupNamesField)
+
+	return frame, nil
+}
+
+func (e *CloudWatchExecutor) executeStartQuery(ctx context.Context, logsClient cloudwatchlogsiface.CloudWatchLogsAPI, parameters *simplejson.Json, timeRange *tsdb.TimeRange) (*cloudwatchlogs.StartQueryOutput, error) {
+	startTime, err := timeRange.ParseFrom()
+	if err != nil {
+		return nil, err
+	}
+
+	endTime, err := timeRange.ParseTo()
+	if err != nil {
+		return nil, err
+	}
+
+	if !startTime.Before(endTime) {
+		return nil, fmt.Errorf("Invalid time range: Start time must be before end time")
```

Comment:
[Error strings should not be capitalized](https://github.com/golang/go/wiki/CodeReviewComments#error-strings):
```suggestion
		return nil, fmt.Errorf("invalid time range: Start time must be before end time")
```